sonic_installer/bootloader/aboot.py

```diff
@@ -126,8 +126,14 @@ def set_next_image(self, image):
         return True
 
     def install_image(self, image_path):
-        run_command("/usr/bin/unzip -od /tmp %s boot0" % image_path)
-        run_command("swipath=%s target_path=/host sonic_upgrade=1 . /tmp/boot0" % image_path)
+        run_command(["/usr/bin/unzip", "-od", "/tmp", image_path, "boot0"])
+        env = os.environ.copy()
+        env.update({
+            'swipath': image_path,
+            'target_path': '/host',
+            'sonic_upgrade': '1'
+        })
+        run_command(["/bin/bash", "/tmp/boot0"], env=env)
```

Comment:
>/bin/bash [](http://example.com/codeflow?start=22&length=9)

Will `.` work? No need to assume `bash`.

Reply:
Or just run boot0 as command.

Reply:
AFAIK `boot0` won't be executable so it will need to be executed via an interpreter.
`.` will not work as it's the equivalent to `source` which just runs the script in the current shell environment.
If anything `/bin/bash` should be replaced by `/bin/sh` to preserve the old behavior.

Reply:
Is there any difference b/w running `/tmp/boot0` using `/bin/bash` and `/bin/sh`? The goal of this PR is to remove the use of `shell=True`, if remain using `/bin/sh` then I think we cannot achieve this goal as the shell is invoked explicitly via `shell=True`.

Reply:
In this case, `shell=True` is expected. So we can keep original code, and add comment to suppress semgrep warning.